Doc/tutorial/errors.rst

```diff
@@ -147,10 +147,52 @@ For example, the following code will print B, C, D in that order::
 Note that if the *except clauses* were reversed (with ``except B`` first), it
 would have printed B, B, B --- the first matching *except clause* is triggered.
 
-All exceptions inherit from :exc:`BaseException`, and so it can be used to serve
-as a wildcard. Use this with extreme caution, since it is easy to mask a real
-programming error in this way!  It can also be used to print an error message and
-then re-raise the exception (allowing a caller to handle the exception as well)::
+When an exception occurs, it may have associated values, also known as the
+exception's *arguments*. The presence and types of the arguments depend on the
+exception type.
+
+The *except clause* may specify a variable after the exception name.  The
+variable is bound to the exception instance which typically has an ``args``
+attribute that stores the arguments. For convenience, builtin exception
+types define :meth:`__str__` to print all the arguments without explicitly
+accessing ``.args``.  ::
+
+   >>> try:
+   ...     raise Exception('spam', 'eggs')
+   ... except Exception as inst:
+   ...     print(type(inst))    # the exception instance
+   ...     print(inst.args)     # arguments stored in .args
+   ...     print(inst)          # __str__ allows args to be printed directly,
+   ...                          # but may be overridden in exception subclasses
+   ...     x, y = inst.args     # unpack args
+   ...     print('x =', x)
+   ...     print('y =', y)
+   ...
+   <class 'Exception'>
+   ('spam', 'eggs')
+   ('spam', 'eggs')
+   x = spam
+   y = eggs
+
+The exception's :meth:`__str__` output is printed as the last part ('detail')
+of the message for unhandled exceptions.
+
+:exc:`BaseException` is a common base class of all exceptions, including a few
+which are not typically handled, because they are used to indicate that the
+program should terminate. They include :exc:`SystemExit` which is raise by
+``sys.exit()`` and :exc:`KeyboardInterrupt` which is raised when a user
+attempts wishes to interrupt the program.
+
+:exc:`Exception` is a subclass of :exc:`BaseException`, which is a common
+superclass of all exceptions, apart from those few special ones that we
+should not typically handle. As such, :exc:`Exception` can be used as a
```

Comment:
This repeats lines 180-184, should probably be adapted before merge (no rush!)

Reply:
180-184 are talking about BaseException, and this is about Exception.

Reply:
Ah, I read that as an apposition `BaseException, which is a common etc`, so it is ambiguous.

Reply:
Ah after re-reading I see the repetition, will fix!